src/utils.jl

```diff
@@ -14,3 +14,24 @@ maybe_only(x::Tuple{T}) where {T} = only(x)
 
 front(itr, n=1) = Iterators.take(itr, length(itr) - n)
 tail(itr) = Iterators.drop(itr, 1)
+
+# Tree utils
+function line_to_tree(line::Vector)
+  if length(line) == 1 && line[1] isa Vector
+    return line[1]
+  end
+  if length(line) <= 2
+    return line
+  end
+  return [line_to_tree(line[1:(end - 1)]), line[end]]
+end
+
+# vectorize a tree
+# example: [[1,2], [3,4]] = [1, 2, 3, 4]
+function vectorize(tree)
```

Comment:
There is `collect(Leaves(tree))` from `AbstractTrees.jl` for this.

Reply:
Done, removed this function